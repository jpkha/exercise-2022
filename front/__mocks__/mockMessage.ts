import {Message} from '../model/api/message';

export const mockMessageEmail: Message = {
  'body': 'Lorem Ipsum #10100 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
  'contact': {
    'email': 'cdavis@gmail.com',
    'firstname': 'Charles',
    'lastname': 'Davis',
    'phone': '0669581508'
  },
  'date': '2021-12-30T11:57:24.496370',
  'id': 10100,
  'read': true,
  'subject': 'Email #10100',
  'type': 'email'
}

export const mockMessagePhone: Message = {
  'body': 'Lorem Ipsum #10100 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
  'contact': {
    'email': 'cdavis@gmail.com',
    'firstname': '',
    'lastname': '',
    'phone': '0669581508'
  },
  'date': '2021-12-30T11:57:24.496370',
  'id': 10100,
  'read': true,
  'subject': 'Phone #10100',
  'type': 'phone'
}

export const mockMessageSms: Message = {
  'body': 'Lorem Ipsum #10100 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
  'contact': {
    'email': 'cdavis@gmail.com',
    'firstname': 'Charles',
    'lastname': 'Davis',
    'phone': '0669581508'
  },
  'date': '2021-12-30T11:57:24.496370',
  'id': 10100,
  'read': false,
  'subject': 'Sms #10100',
  'type': 'sms'
}
