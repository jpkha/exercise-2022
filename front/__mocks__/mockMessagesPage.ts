import {Message} from '../model/api/message';

export const mockMessagesPage: Message[] = [
  {
    "body": "Lorem Ipsum #10100 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "cdavis@gmail.com",
      "firstname": "Charles",
      "lastname": "Davis",
      "phone": "0669581508"
    },
    "date": "2021-12-30T11:57:24.496370",
    "id": 10100,
    "read": true,
    "subject": "Email #10100",
    "type": "email"
  },
  {
    "body": "Lorem Ipsum #10101 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "wjohnson@gmail.com",
      "firstname": "William",
      "lastname": "Johnson",
      "phone": "0684515233"
    },
    "date": "2021-12-30T11:57:24.496402",
    "id": 10101,
    "read": false,
    "subject": "Appel #10101",
    "type": "phone"
  },
  {
    "body": "Lorem Ipsum #10102 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "jwilliams@gmail.com",
      "firstname": "John",
      "lastname": "Williams",
      "phone": "0637498492"
    },
    "date": "2022-01-05T11:57:24.496425",
    "id": 10102,
    "read": false,
    "subject": "Appel #10102",
    "type": "phone"
  },
  {
    "body": "Lorem Ipsum #10103 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "jjones@gmail.com",
      "firstname": "James",
      "lastname": "Jones",
      "phone": "0645436220"
    },
    "date": "2021-12-24T11:57:24.496447",
    "id": 10103,
    "read": false,
    "subject": "Email #10103",
    "type": "email"
  },
  {
    "body": "Lorem Ipsum #10104 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "mdavis@gmail.com",
      "firstname": "Michael",
      "lastname": "Davis",
      "phone": "0681731880"
    },
    "date": "2021-12-22T11:57:24.496468",
    "id": 10104,
    "read": false,
    "subject": "Appel #10104",
    "type": "phone"
  },
  {
    "body": "Lorem Ipsum #10105 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "jtaylor@gmail.com",
      "firstname": "John",
      "lastname": "Taylor",
      "phone": "0699491953"
    },
    "date": "2021-12-30T11:57:24.496490",
    "id": 10105,
    "read": false,
    "subject": "Email #10105",
    "type": "email"
  },
  {
    "body": "Lorem Ipsum #10106 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "djackson@gmail.com",
      "firstname": "David",
      "lastname": "Jackson",
      "phone": "0637389545"
    },
    "date": "2022-01-08T11:57:24.496512",
    "id": 10106,
    "read": true,
    "subject": "Email #10106",
    "type": "email"
  },
  {
    "body": "Lorem Ipsum #10107 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "wsmith@gmail.com",
      "firstname": "William",
      "lastname": "Smith",
      "phone": "0690623489"
    },
    "date": "2021-12-26T11:57:24.496533",
    "id": 10107,
    "read": false,
    "subject": "Appel #10107",
    "type": "phone"
  },
  {
    "body": "Lorem Ipsum #10108 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "rdavis@gmail.com",
      "firstname": "Richard",
      "lastname": "Davis",
      "phone": "0685823430"
    },
    "date": "2021-12-19T11:57:24.496555",
    "id": 10108,
    "read": false,
    "subject": "SMS #10108",
    "type": "sms"
  },
  {
    "body": "Lorem Ipsum #10109 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "contact": {
      "email": "wwilliams@gmail.com",
      "firstname": "William",
      "lastname": "Williams",
      "phone": "0659206302"
    },
    "date": "2022-01-04T11:57:24.496576",
    "id": 10109,
    "read": true,
    "subject": "Email #10109",
    "type": "email"
  }
];
