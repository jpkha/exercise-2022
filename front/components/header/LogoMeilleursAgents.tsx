import Link from 'next/link';
import Image from 'next/image';
import styled from 'styled-components';

export const LogoMeilleursAgents = ({realtorsId}: {realtorsId: string}) => {
  const LogoContainer = styled.div`
  flex: 1 0 auto;
`
  const Logo = styled.h1`
  height: 37px;
  max-width: 133px;
  margin-left: 12px;
  position: relative;

  img {
    min-width: unset;
    max-width: unset;
    width: auto;
  }
`
  return <LogoContainer>
    <Logo>
      <Link href={`/realtors/${realtorsId}`} passHref>
        <a title="Revenir Ã  la messagerie"><Image src="/logo-meilleursagents.svg" layout={'fill'} objectFit={'contain'} loading={'lazy'}
                  alt="logo de Meilleurs agents"/></a>
      </Link>
    </Logo>
  </LogoContainer>
}
