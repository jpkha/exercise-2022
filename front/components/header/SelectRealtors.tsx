import {Realtor} from '../../model/api/realtor';
import {useRouter} from 'next/router';
import {ChangeEvent} from 'react';
import styled from 'styled-components';
import { SrOnly } from '../../styles/utils';

const SrOnlyLabel = styled.label`
  ${SrOnly}
`

export const SelectRealtors = ({realtors}: { realtors: Realtor[] }) => {
  const router = useRouter();

  const handleSelectAgencies = (event: ChangeEvent<HTMLSelectElement>) => {
    event.preventDefault();
    const realtorId = event.target.value;
    router.push(`/realtors/${realtorId}`);
  };

  let selectedRealtor = '';
  if (router.query.realtorsId) {
    selectedRealtor = router.query.realtorsId.toString();
  }

  return <>
    <SrOnlyLabel htmlFor="agency-select">Choississez une agence:</SrOnlyLabel>
    <select value={selectedRealtor} name="agencies" id="agency-select" onChange={handleSelectAgencies}>
      <option value="">-- Choissisez une option--</option>
      {realtors?.map((agency: Realtor) =>
        <option key={agency.id} value={agency.id}> {agency.name.replace(/#/g, '')}</option>)}
    </select>
  </>
}
