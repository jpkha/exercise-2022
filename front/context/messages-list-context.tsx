import {createContext, useContext, useState} from 'react';

export const MessagesListContext = createContext({
  messages: [],
  pages: 1,
  resetMessages: () => {
  },
  setMessages: () => {}
});

export function MessagesListProvider({children, selectedRealtor}) {
  const [messages, setMessages] = useState([]);
  const [pages, setPages] = useState(1);

  const resetMessages = () => {
    setMessages([]);
    setPages(1);
  }
  const value = {
    messages,
    pages,
    setMessages,
    resetMessages,
  }

  return (
    <MessagesListContext.Provider value={value}>
      {children}
    </MessagesListContext.Provider>
  )
}

export function useMessagesListContext(): any {
  return useContext(MessagesListContext);
}
