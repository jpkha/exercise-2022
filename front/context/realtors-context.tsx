import {createContext, useContext, useState} from 'react';
import {Realtor} from '../model/api/realtor';

const realtorsContextDefaultValue = {
  realtors: [],
  realtorHasReadOneMessage: (realtorId: string) => {
  }
};

export const RealtorsContext = createContext(realtorsContextDefaultValue);


export const RealtorsProvider = ({children, realtors: realtorsRes}) => {

  const [realtors, setRealtors] = useState(realtorsRes);
  const realtorHasReadOneMessage = (realtorId: string) => {
    const index = realtors.findIndex((realtor: Realtor) => realtor.id.toString() === realtorId);
    realtors[index].unread_messages -= 1;
  }

  const value = {
    realtors,
    setRealtors,
    realtorHasReadOneMessage
  }
  return <RealtorsContext.Provider value = {value}> {children} </RealtorsContext.Provider>

}

export function useRealtorsContext() {
  return useContext(RealtorsContext);
}
