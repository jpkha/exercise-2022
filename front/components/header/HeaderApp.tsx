import styled from 'styled-components'
import {Realtor} from '../../model/api/realtor'
import {SelectRealtors} from './SelectRealtors';
import {useRouter} from 'next/router';
import {greyBackgroundBoxColor, primaryColor} from '../../styles/variables';
import {LogoMeilleursAgents} from './LogoMeilleursAgents';
import {useContext} from 'react';
import {RealtorsContext} from '../../context/realtors-context';

const HeaderContainer = styled.header`
  height: 60px;
  display: flex;
  flex: 0 0 auto;
  align-items: center;
  box-shadow: 0 2px 4px 0 rgb(58 58 58 / 30%);
  position: relative;
  z-index: 1;
`


const UnreadMessageContainer = styled.div`
  flex: 0 0 auto;
  color: white;
  background-color: ${({emptyBox}) => emptyBox ? greyBackgroundBoxColor : primaryColor};
  margin: 0 12px;
  padding: 5px 12px;
  margin: 0 12px;
  border-radius: 8px;

  > i {
    padding-right: 10px;
  }
`
export const HeaderApp = () => {
  const router = useRouter();
  const {realtors} = useContext(RealtorsContext);
  const realtorsId = router.query.realtorsId?.toString();
  const selectedRealtor = realtors.find((realtor: Realtor) => realtorsId === realtor.id.toString());
  return (
    <HeaderContainer>
      <LogoMeilleursAgents realtorsId={realtorsId}/>
      {selectedRealtor &&
      <UnreadMessageContainer emptyBox={selectedRealtor.unread_messages === 0} data-testid="unread-message">
          <i className="mypro-icon mypro-icon-inbox"></i>{selectedRealtor.unread_messages}
      </UnreadMessageContainer>
      }
      <SelectRealtors realtors={realtors}/>
    </HeaderContainer>
  )
}
