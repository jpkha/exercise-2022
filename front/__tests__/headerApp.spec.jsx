// __tests__/headerApp.spec.jsx

/**
 * @jest-environment jsdom
 */

import React from 'react'
import {fireEvent, render, screen} from '@testing-library/react'
import {HeaderApp} from '../components/header/HeaderApp';
import * as nextRouter from 'next/router';
import {mockRealtors} from '../__mocks__/mockRealtors';
import {RealtorsProvider} from '../context/realtors-context';

describe('HeaderApp', () => {
    const push = jest.fn();
    let tree;
    beforeEach(() => {
        nextRouter.useRouter = jest.fn();
        nextRouter.useRouter.mockImplementation(() => ({
            query: {realtorsId: '101'},
            push
        }));
        tree = render(
            <RealtorsProvider realtors={mockRealtors}>
                <HeaderApp/>
            </RealtorsProvider>
        )
    })
    it('renders correctly a heading', () => {
        expect(tree).toMatchSnapshot();
    })

    it('renders a heading', () => {
        const heading = screen.getByRole('heading', {
            name: 'Revenir Ã  la messagerie',
        })
        expect(heading).toBeInTheDocument()
        const logo = screen.getByRole('img', {
            name: 'logo de Meilleurs agents',
        })
        expect(logo).toBeInTheDocument()
        const selectOption = screen.getByTestId('select-realtors');
        expect(selectOption).toBeInTheDocument()

    })

    it('onSelect agency, it should push a new route', () => {
        const selectOption = screen.getByTestId('select-realtors');
        expect(selectOption).toBeInTheDocument()
        const options = screen.getAllByRole('option');
        expect(options.length).toBe(1 + mockRealtors.length);
        fireEvent.change(selectOption, {target: {value: 101}});
        expect(options[0].selected).toBeFalsy();
        expect(options[1].selected).toBeTruthy();
        expect(options[2].selected).toBeFalsy();
        expect(options[3].selected).toBeFalsy();
        expect(push).toHaveBeenCalledWith('/realtors/101');

    })
})
