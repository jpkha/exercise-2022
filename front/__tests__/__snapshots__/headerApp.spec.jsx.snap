// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderApp renders correctly a heading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <header
        class="sc-jRQBWg bPQgQj"
      >
        <div
          class="sc-hKwDye hGxwYb"
        >
          <h1
            class="sc-eCImPb jtYecN"
          >
            <a
              href="/realtors/101"
              title="Revenir à la messagerie"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="logo de Meilleurs agents"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </a>
          </h1>
        </div>
        <div
          class="sc-gKclnd bQLyGH"
          data-testid="unread-message"
        >
          <i
            class="mypro-icon mypro-icon-inbox"
          />
          55
        </div>
        <div
          class="sc-gsDKAQ feabsB"
        >
          <label
            class="sc-bdvvtL bmZBfL"
            for="agency-select"
          >
            Choississez une agence:
          </label>
          <select
            class="sc-dkPtRN iwAvwK"
            data-testid="select-realtors"
            id="agency-select"
            name="agencies"
          >
            <option
              value=""
            >
              -- Choissisez une option--
            </option>
            <option
              value="101"
            >
               Agence 101
            </option>
            <option
              value="102"
            >
               Agence 102
            </option>
            <option
              value="103"
            >
               Agence 103
            </option>
          </select>
        </div>
      </header>
       
    </div>
  </body>,
  "container": <div>
     
    <header
      class="sc-jRQBWg bPQgQj"
    >
      <div
        class="sc-hKwDye hGxwYb"
      >
        <h1
          class="sc-eCImPb jtYecN"
        >
          <a
            href="/realtors/101"
            title="Revenir à la messagerie"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="logo de Meilleurs agents"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </a>
        </h1>
      </div>
      <div
        class="sc-gKclnd bQLyGH"
        data-testid="unread-message"
      >
        <i
          class="mypro-icon mypro-icon-inbox"
        />
        55
      </div>
      <div
        class="sc-gsDKAQ feabsB"
      >
        <label
          class="sc-bdvvtL bmZBfL"
          for="agency-select"
        >
          Choississez une agence:
        </label>
        <select
          class="sc-dkPtRN iwAvwK"
          data-testid="select-realtors"
          id="agency-select"
          name="agencies"
        >
          <option
            value=""
          >
            -- Choissisez une option--
          </option>
          <option
            value="101"
          >
             Agence 101
          </option>
          <option
            value="102"
          >
             Agence 102
          </option>
          <option
            value="103"
          >
             Agence 103
          </option>
        </select>
      </div>
    </header>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
